network/p2p/middleware/middleware.go

```diff
@@ -512,6 +518,24 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 			return
 		}
 
+		channel := channels.Channel(msg.ChannelID)
+		topic := channels.TopicFromChannel(channel, m.rootBlockID)
+
+		// msg type is not guaranteed to be correct since it is set by the client
+		_, what, err := codec.InterfaceFromMessageCode(msg.Payload[0])
```

Comment:
Looks like `what` is only used within the `if !m.libP2PNode.HasSubscription(topic) {` scope, can you move it there to avoid the unnecessary lookup?

Reply:
https://github.com/onflow/flow-go/pull/3490/commits/1a315656d8086d2b1c54648f515f37da8644f062